pkg/scheduler/api/job_info.go

```diff
@@ -504,6 +526,39 @@ func (ji *JobInfo) GetMinResources() *Resource {
 	return NewResource(*ji.PodGroup.Spec.MinResources)
 }
 
+// Get the total resources of tasks whose pod is scheduling gated
+// By definition, if a pod is scheduling gated, it's status is Pending
+func (ji *JobInfo) GetSchGatedPodResources() *Resource {
+	res := EmptyResource()
+	for _, task := range ji.Tasks {
+		if task.SchedulingGated() {
+			res.Add(task.Resreq)
+		}
+	}
+	return res
+}
+
+// Deduct resources of scheduling gated pod from Resource res;
+// If resource is less than gated resources, return zero;
+// Gated resources might be more than minResource when total rep of each task > minAvailable
+func (ji *JobInfo) DeductSchGatedResources(res *Resource) *Resource {
+	result := res.Clone()
```

Comment:
We can check whether res Is zero to have a shortcut here, because in most case task is not gated.